src/game_interpreter.cpp

```diff
@@ -388,14 +421,15 @@ void Game_Interpreter::Update(bool reset_loop_count) {
 			continue;
 		}
 
-		// FIXME?
-		// After calling SkipTo this index++ will skip execution of e.g. END.
-		// This causes a different timing because loop_count reaches 10000
-		// faster then Player does.
-		// No idea if any game depends on this special case.
 		// Note: In the case we executed a CallEvent command, be sure to
 		// increment the old frame and not the new one we just pushed.
-		_state.stack[current_frame_idx].current_command++;
+		frame = &_state.stack[current_frame_idx];
+
+		// Only do auto increment if the command didn't manually 
+		// change the index.
```

Comment:
Under which cases does this happen?

Reply:
Any commands that need to manipulate the `index` internally. The show message is one complex example.

Loop commands are another